howso/engine/trainee.py

```diff
@@ -522,19 +515,15 @@ def set_default_features(
             self._default_context_features = None
         self.update()
 
-    def set_metadata(self, metadata: Optional[Dict[str, Any]]) -> None:
+    def set_metadata(self, metadata: Optional[Dict[str, Any]]):
```

Comment:
I wish this was MutableMapping instead of dict. Also, the docstring for this should be "Map of string to anything, optional"